### PR TITLE
Decision making statement/it-academy/M-JC1-69-21/andreidrozd/done tasks 1-10, 12-14

### PR DESCRIPTION
**Pull request by andreidrozd**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

**Comment by andreidrozd** [comment]
solution: Done task13

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task01.java [hunk 1]
```diff
@@ -36,8 +36,6 @@ public static void main(String[] args) {
      * </ul>
      */
     static boolean isTopPlayer(int sizeTop, int positionPlayerByDamage) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return false;
+        return (sizeTop >= positionPlayerByDamage);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task01.java:39]:
```suggestion
        return sizeTop >= positionPlayerByDamage;
```

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task02.java [hunk 2]
```diff
@@ -35,8 +35,18 @@ public static void main(String[] args) {
...
+        }
+        if (hour >= 12 && hour <= 17) {
+            return ("Good day");
+        }
+        if (hour >= 18 && hour <= 23) {
+            return ("Good evening");
+        }
+        if (hour >= 0 && hour <= 5) {
+            return ("Good night");
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task02.java:49]: This isn't effective.

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task03.java [hunk 3]
```diff
@@ -35,8 +35,23 @@ public static void main(String[] args) {
...
+                return "ORANGE";
+            }
+        if (percentageRatio >= 50)
+            if (percentageRatio < 75) {
+                return "YELLOW";
+            }
+        if (percentageRatio >= 75)
+            if (percentageRatio <= 100) {
+                return "GREEN";
+            }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task03.java:54]: This is difficult

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task04.java [hunk 4]
```diff
@@ -29,8 +29,23 @@ public static void main(String[] args) {
      * @return день недели на английском языке (UPPER CASE)
      */
     static String getDayOfWeekNumber(byte dayOfWeekNumber) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        switch (dayOfWeekNumber) {
+            case 1:
+                return ("MONDAY");
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task04.java:34]:
```suggestion
                return "MONDAY";
```

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task06.java [hunk 5]
```diff
@@ -39,8 +39,15 @@ public static void main(String[] args) {
...
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
+        int scoreFirstPlayer = tanksKilledFirstPlayer * KILL_POINT;
+        int scoreSecondPlayer = tanksKilledSecondPlayer * KILL_POINT;
+        if (tanksKilledFirstPlayer > tanksKilledSecondPlayer) {
+            return ("Player 1: " + (scoreFirstPlayer + BONUS_POINT) + ". " + "Player 2: " + scoreSecondPlayer);
+        } else if (scoreFirstPlayer < scoreSecondPlayer) {
+            return ("Player 1: " + scoreFirstPlayer + ". " + "Player 2: " + (scoreSecondPlayer + BONUS_POINT));
+        } else if (scoreFirstPlayer == scoreSecondPlayer) {
+            return ("Player 1: " + scoreFirstPlayer + ". " + "Player 2: " + scoreSecondPlayer);
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task06.java:50]: Violation of DRY principle

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task08.java [hunk 6]
```diff
@@ -38,8 +38,26 @@ public static void main(String[] args) {
...
+                case "SAINT":
+                    return damage / 2;
+                case "ANIMAL":
+                    return damage;
+                case "HUMANOID":
+                    return damage;
+                case "PLANT":
+                    return damage;
+                case "GHOST":
+                    return damage;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task08.java:56]:
```suggestion
                case "ANIMAL":
                case "HUMANOID":
                case "PLANT":
                case "GHOST":
                    return damage;
```

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task10.java [hunk 7]
```diff
@@ -35,8 +35,26 @@ public static void main(String[] args) {
...
+        }
+        if (firstFuelReserves == secondFuelReserves) {
+            if (firstFuelReserves < thirdFuelReserves)
+                return 1;
+        }
+        if (secondFuelReserves == thirdFuelReserves) {
+            if (secondFuelReserves < firstFuelReserves)
+                return 2;
+        }
+        return 1;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task10.java:58]: This is difficult

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task11.java [hunk 8]
```diff
@@ -38,8 +38,15 @@ public static void main(String[] args) {
      * @return <code>latin</code>/<code>cyrillic</code>/<code>digit</code>/<code>undefined</code>
      */
     static String getSymbolType(char symbol) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
-        return null;
+        if (Character.isDigit(symbol)) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task11.java:41]: What is `Character`?

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task12.java [hunk 9]
```diff
@@ -45,8 +45,79 @@ public static void main(String[] args) {
...
+                        if (day <= 19)
+                            if (month == 2)
+                                return "Water-bearer";
+            if (day >= 20)
+                if (month == 2)
+                    return "Fish";
+                        if (day <= 20)
+                            if (month == 3)
+                                return "Fish";
+        }    return null;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task12.java:121]: - This is difficult.
- Violations of Code Convention

File: src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task14.java [hunk 10]
```diff
@@ -46,8 +46,31 @@ public static void main(String[] args) {
      * @return название года по китайскому календарю
      */
     static String getZodiacYearName(int day, int month, int year) {
-        //TODO
-        // Код, решающий задачу пишем ниже, при этом используя параметры метода
+        if (day > 0 && month > 0)
+            if ((year % 12) == 0)
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/decision_making_statement/Task14.java:50]:
```suggestion
            if (year % 12 == 0)
```

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix the following problems

**Pull request closed without merging**